### PR TITLE
Add Totals to CartItemInterface

### PR DESCRIPTION
**Pull request by pmclain**
### Fixed Issues (if relevant)
1. magento/graphql-ce#633

### Manual testing scenarios (*)
1. Create empty cart
2. Add cart items
3. Query cart and request item totals
```graphql
{
  cart(cart_id: "{{cartId}}") {
    items {
      prices {
        price {
          value
          currency
        }
        row_total {
          value
          currency
        }
        row_total_including_tax {
          value
          currency
        }
      }
    }
    prices {
      grand_total {
        value
        currency
      }
      subtotal_including_tax {
        value
        currency
      }
      subtotal_excluding_tax {
        value
        currency
      }
      subtotal_with_discount_excluding_tax {
        value
        currency
      }
      applied_taxes {
        label
        amount {
          value
          currency
        }
      }
    }
  }
}
```

### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All new or changed code is covered with unit/integration tests (if applicable)
 - [x] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

**Review by vpodorozh — APPROVED** [review]

**Comment by TomashKhamlai** [comment]
@pmclain, could you please merge mainline to this PR?

**Comment by pmclain** [comment]
@TomashKhamlai mainline merged https://github.com/magento/graphql-ce/pull/664/commits/d26dbbc5e8c16a02a91b6628f50b24faecd17f69

**Comment by lenaorobei** [comment]
@magento run all tests

**Comment by lenaorobei** [comment]
@paliarush please approve schema changes.

File: app/code/Magento/QuoteGraphQl/etc/schema.graphqls [hunk 1]
```diff
@@ -300,9 +300,16 @@ type VirtualCartItem implements CartItemInterface @doc(description: "Virtual Car
 interface CartItemInterface @typeResolver(class: "Magento\\QuoteGraphQl\\Model\\Resolver\\CartItemTypeResolver") {
     id: String!
     quantity: Float!
+    prices: CartItemPrices @resolver(class: "\\Magento\\QuoteGraphQl\\Model\\Resolver\\CartItemPrices")
```
Comment (paliarush) [inline app/code/Magento/QuoteGraphQl/etc/schema.graphqls:303]: Current implementation is based on the assumption that prices can be extracted from the `product` field. If this assumption is incorrect, please explain.

**Comment by paliarush** [comment]
@joni-jones please review this PR since quote is your component.

File: app/code/Magento/QuoteGraphQl/etc/schema.graphqls (hunk 1 above)
Comment (pmclain) [reply app/code/Magento/QuoteGraphQl/etc/schema.graphqls:303]: I've worked on a number of sites with external integrations powering cart pricing. In these stores, most B2B with contract pricing, there can be a disconnect between the product price and the cart item price.

**Review by YevSent — APPROVED** [review]

**Review by lenaorobei — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @lenaorobei, thank you for the review.
[ENGCOM-5576](https://jira.corp.magento.com/browse/ENGCOM-5576) has been created to process this Pull Request
:eight_spoked_asterisk: @lenaorobei, could you please add one of the following labels to the Pull Request?

| Label | Description |
|--------------------------------|-----------------------------------------------------------------|
| ```Auto-Tests: Covered``` | All changes in Pull Request is covered by auto-tests |
| ```Auto-Tests: Not Covered``` | Changes in Pull Request requires coverage by auto-tests |
| ```Auto-Tests: Not Required``` | Changes in Pull Request does not require coverage by auto-tests |

**Comment by keharper** [comment]
Schema changes documented in 2.3.4 as a result of work done by the internal development team.

**Pull request merged**